cisc/cisc.go

```diff
@@ -547,13 +548,52 @@ func kvAdd(c *cli.Context) error {
 	value := c.Args().Get(1)
 	prop := id.GetProposed()
 	prop.Storage[key] = value
+	return addKv(c, cfg, id, prop)
+}
+
+// kvAddCsv reads the input CSV file, lookup its key column and inputs all the
+// // csv file at once to the skipchain. The key is given from the "column" args
+// and the value is the whole full row, unmodified.
+func kvAddCsv(c *cli.Context) error {
+	cfg := loadConfigOrFail(c)
+	if c.NArg() != 1 {
+		return errors.New("Missing argument: csv file")
+	}
+	id, err := cfg.findSC(c.Args().Get(1))
+	if err != nil {
+		scList(c)
+		return errors.New("Please give skipchain-id")
+	}
+	file := c.Args().First()
+	fd, err := os.Open(file)
+	log.ErrFatal(err)
+	reader := csv.NewReader(fd)
```

Comment:
This would be more efficient if you fed a bufio.Reader to it (https://godoc.org/bufio#NewReader). I don't know if the expected scale justifies this.

Reply:
See my comment below. it's related in my opinion. The way we read the file and the way we propose new data should be considered.